services/QuillLMS/app/services/classlink_integration/school_persister.rb

```diff
@@ -27,12 +27,8 @@ def run
     end
 
     private def school
-      return unless nces_id
-
-      school = School.find_by(nces_id:)
-      return unless school
-
-      school.tap { |s| s.update!(name:, district_id:) }
+      school = nces_id ? School.find_by(nces_id:, district_id:) : School.find_by(name:, district_id:)
```

Comment:
Note: per partnerships onboarding policy, these `School` records will have been created by the time this rostering code is executed.